apps/ledger-live-mobile/src/components/DeviceAction/index.tsx

```diff
@@ -35,29 +35,52 @@ type Props<R, H, P> = {
   onResult?: (_: any) => Promise<void> | void;
   onError?: (_: any) => Promise<void> | void;
   renderOnResult?: (_: P) => React.ReactNode;
-  action: Action<R, H, P>;
+  action?: Action<R, H, P>;
+  status?: any;
   request?: R;
   device: Device;
+  payload?: any;
   onSelectDeviceLink?: () => void;
   analyticsPropertyFlow?: string;
 };
+
 export default function DeviceAction<R, H, P>({
   action,
   request = null,
   device: selectedDevice,
+  ...props
+}: Props<R, H, P>): JSX.Element {
+  const status: any = action.useHook(selectedDevice, request);
+  const payload = action.mapResult(status);
+
+  return (
+    <DeviceActionDefaultRendering
+      device={selectedDevice}
+      status={status}
+      request={request}
+      payload={payload}
+      {...props}
+    />
+  );
+}
+
+export function DeviceActionDefaultRendering({
   onResult,
   onError,
+  device: selectedDevice,
   renderOnResult,
   onSelectDeviceLink,
   analyticsPropertyFlow = "unknown",
-}: Props<R, H, P>) {
+  status,
+  request,
+  payload,
+}: Props<R, H, P>): JSX.Element {
```

Comment:
You need to define `R, H, P` by setting `DeviceActionDefaultRendering` as a generic:

```TS
export function DeviceActionDefaultRendering<R, H, P>({ ... })
```

Reply:
Also: not sure about the returned type: there are several `return null`, the easy TS fix would be to have: `JSX.Element | null` as the return type